### ISSUE TITLE
False positive with `:host` and `:host()`

### ISSUE THREAD

**Issue by veyndan** [csstools/stylelint-prefer-nesting#23]
The plugin gives a false positive that `:host()` can be nested within `:host` when it can't.

### Source

```css
:host {
    background: hotpink;
}

:host(.foo) {
    background: rebeccapurple;
}
```

### Expected

The file to remain the same.

### Actual

```css
:host {
    background: hotpink;

    &(.foo) {
        background: rebeccapurple;
    }
}
```
